UTNAK/kepler16b-using-imce-vocabulary
# 7 execute simulation code on a cicd pipeline

## Checklist before submitting a merge request


## Description of contribution

I added and tested CI/CD pipeline using github actions.

## Testing performed


<img width="845" alt="image" src="https://github.com/UTNAK/kepler16b-using-imce-vocabulary/assets/29029659/a47b41a2-e91d-428d-8a6b-81ce8270618c">


## How to Test Expected functionality changes

1. `git clone https://github.com/UTNAK/kepler16b-using-imce-vocabulary.git`
2. switch to the branch `7-execute-simulation-code-on-a-cicd-pipeline`
3. open `src/analysis/analysisShell.qmd` and add some text
4. commit and pull. 
5. After that, the github actions automatically runs CI/CD pipeline
6. Go to [Actions](https://github.com/UTNAK/kepler16b-using-imce-vocabulary/actions)
7. Check the workflow is running. It takes 10 minutes typically.
8. Check the status of build and deploy task is succeeded.
9. Go to [deployed pages](https://utnak.github.io/kepler16b-using-imce-vocabulary/)
10. Check the timestamp is updated

<img width="969" alt="image" src="https://github.com/UTNAK/kepler16b-using-imce-vocabulary/assets/29029659/b033d976-47f9-4791-bbc1-6a59dd82be4f">


11. Check the analysis results of a [quarto notebook](https://utnak.github.io/kepler16b-using-imce-vocabulary/analysisShell.html)
12. Check the analysis results of a [jupyter notebook](https://utnak.github.io/kepler16b-using-imce-vocabulary/analysisJupyter.html)


- [x] I have added the steps needed to test these changes

## Additional context

[If needed, you may add additional context]
